ql/src/java/org/apache/hadoop/hive/ql/exec/DDLTask.java

```diff
@@ -1159,6 +1160,12 @@ private int renamePartition(Hive db, RenamePartitionDesc renamePartitionDesc) th
       return 0;
     }
 
+    String names[] = Utilities.getDbTableName(tableName);
```

Comment:
Exchange partition does drop+add internally which does;t have the issue like rename. Rename looks for source table but add/drop don't care. So, no problem.